src/asl.ts

```diff
@@ -2918,6 +2934,285 @@ export class ASL {
     );
   }
 
+  public evalDecl(
+    decl: VariableDecl | ParameterDecl,
+    initialValue?: ASLGraph.Output
+  ): ASLGraph.SubState | ASLGraph.NodeState | undefined {
+    const getState = () => {
+      if (initialValue) {
+        return this.evalAssignment(decl.name, initialValue);
+      } else if (decl.initializer === undefined) {
+        return undefined;
+      }
+
+      return this.evalExprToSubState(decl.initializer, (exprOutput) => {
+        return this.evalAssignment(decl.name, exprOutput);
+      });
+    };
+    const state = getState();
```

Comment:
Would prefer to just inline `(() => {})()` or a ternary. Const functions drive be nuts because code reads backwards